alacritty/src/input.rs

```diff
@@ -394,17 +394,24 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
                 ClickState::TripleClick
             }
             _ => {
-                // Don't launch URLs if this click cleared the selection
-                self.ctx.mouse_mut().block_url_launcher = !self.ctx.selection_is_empty();
+                // Only clear selection if LMB is pressed and
+                // if either SHIFT is pressed or mouse mode is disabled
```

Comment:
This comment is basically just as verbose as the code itself, so it doesn't really improve the situation.

It's also not accurate since this is also about starting a selection, not only about clearing it.